mta-archive-builder (#3)
SAP/devops-docker-images: mta-archive-builder/README.md

# Multitarget Application Archive Builder
The multitarget application archive builder is a command-line tool that packages a multitarget application into a deployable archive (MTAR). For full documentation please visit the [SAP Help Portal](https://help.sap.com/viewer/58746c584026430a890170ac4d87d03b/Cloud/en-US/ba7dd5a47b7a4858a652d15f9673c28d.html).

This image can be used to build SAP Multitarget Applications (MTA) containing Java and Node.js modules.

# How to use this image

On a linux machine you can run 

> `docker run --rm -v `pwd`:/project -it mta-archive-builder:latest mtaBuild --version`

Reviewer: Why do we mount here? this example only prints the version.

Author: for `mta --version` there is no need to mount a folder into `/project`. This is more for didactical reason. But in the strong sense we can omit this.

> `docker run --rm -v `pwd`:/project -it mta-archive-builder:latest mtaBuild --version`

Reviewer: Why `-it`? The example should run without it.

```suggestion
`docker run --rm mta-archive-builder mtaBuild --version`
```

Author: I think `-it` is not strongly required

This will execute the MTA archive builder and print its version information.

> `docker run --rm -v `pwd`:/project -it mta-archive-builder:latest mtaBuild --mtar dummy.mtar --build-target NEO build`

Reviewer: Not sure about the `:latest`, in my opinion its an antipattern.
```suggestion
`docker run --rm -v `pwd`:/project mta-archive-builder mtaBuild --mtar dummy.mtar --build-target NEO build`
```

Author: IMO `<imageName>` and `<imageName>:latest` are fully equivalent. Having the `:latest` is more transparent.

This will build an `mtar` file for SAP Cloud Platform (Neo). The folder containing the project needs to be mounted into the image at `/project`.

# How to build this image

`docker build -t mta-archive-builder .`

## This image provides:

- SAP Multitarget Application Archive Builder
- Nodejs
- SAP registry (`@sap:registry https://npm.sap.com`) contained in global node configuration.
- Maven

> MTA delegates to other build tools. This image provides node and maven, so MTA can delegate

Reviewer: ```suggestion
The MTA Archive Builder delegates module builds to other build tools. This image provides nodejs and maven, so the archive builder can delegate
```
